storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -4918,6 +4915,7 @@ xtrabackup_backup_func(void)
 	mutex_exit(&log_sys->mutex);
 
 	/* copy log file by current position */
+	mutex_create(LATCH_ID_XTRA_MDL_TAKEN_MUTEX, &mdl_taken_mutex);
 	if(xtrabackup_copy_logfile(checkpoint_lsn_start, FALSE))
 		exit(EXIT_FAILURE);
 
```

Comment:
May be it is better if we do MDL locking before we call xb_load_tablesapces() [which determines the list of files to copy]

This will also help to get rid of this mutex and syncronization with redo thread.

Reply:
@satya-bodapati - Done. Now we:

1. Execute MDL on all tables via `mdl_lock_tables` 
2. Do the first redo log copy phase, so we are caught up with everything that happened before the backup start
3. Change `mdl_taken` to `true`
4. Start `log_copying_thread` - this thread should abort the backup if it finds `MLOG_INDEX_LOAD`

Let me know if this will be sufficient or if we want to go with the approach of validating each scan SN